okhttp/src/main/java/okhttp3/internal/platform/Platform.kt

```diff
@@ -196,6 +196,12 @@ open class Platform {
         return "OpenJSSE" == preferredProvider
       }
 
+    private val isBouncyCastlePreferred: Boolean
+      get() {
+        val preferredProvider = Security.getProviders()[0].name
```

Comment:
I think this is a pretty good policy for Conscrypt because you’re probably only installing it to get access to it’s HTTPS provider. But for BouncyCastle I suspect people install it to fill gaps in the built-in JSSE. In particular, generating private keys!

So I’d prefer BC to require an explicit opt-in.

Reply:
What would that look like? We don't have a public API for this?  System property?

The other consideration is whether to refactor Platform to use the SocketAdapter model everywhere and support all modes at once.

Reply:
Supporting all modes at once would be nice. As we do it we’re potentially risking locking in the wrong platform at application start-up.

Reply:
For now, I’d prefer a system property. I’m particularly anxious about users upgrading to 4.4 and getting BouncyCastle unexpectedly.